hazelcast/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -167,6 +177,14 @@ void setJobStatus(JobStatus jobStatus) {
         setJobStatus(jobStatus, null, false);
     }
 
+    RawJobMetrics getJobMetrics() {
+        return jobMetrics;
```

Comment:
MasterContext and MasterJobContext have 1:1 relationship. Why split the metrics?

BTW: did you implement:
> It should be also persisted after job finishes (if metrics persistence is enabled)